src/Private/Helpers.ps1

```diff
@@ -3156,4 +3144,170 @@ function Test-PodePlaceholders {
+                } else {
+                    '{}'
+                }
+                break
+            }
+            'Boolean' {
+                "$(&{
+                            if ($InputObject -eq $true) { 'true' }
+                            else { 'false' }
+                        })"
+            }
+            'string' {
+                $String = "$InputObject"
+                if (($string -match '[\r\n]' -or $string.Length -gt 80) -and !$string.StartsWith('http')) {
+                    # right, we have to format it to YAML spec.
+                    $folded = ">`n" # signal that we are going to use the readable 'newlines-folded' format
+                    $string.Split("`n") | ForEach-Object {
+                        $workingString = $_ -replace '\r$'
+                        $length = $_.Length
+                        $IndexIntoString = 0
+                        $wrap = 80
+                        while ($length -gt $IndexIntoString + $Wrap) {
+                            $BreakPoint = $wrap
+                            $earliest = $workingString.Substring($IndexIntoString, $wrap).LastIndexOf(' ')
+                            $latest = $workingString.Substring($IndexIntoString + $wrap).IndexOf(' ')
+                            if (($earliest -eq -1) -or ($latest -eq -1)) {
+                                $BreakPoint = $wrap
+                            } elseif ($wrap - $earliest -lt ($latest)) {
+                                $BreakPoint = $earliest
+                            } else {
+                                $BreakPoint = $wrap + $latest
+                            }
+
+                            if (($wrap - $earliest) + $latest -gt 30) {
+                                $BreakPoint = $wrap # in case it is a string without spaces
+                            }
+
+                            $folded += $padding + $workingString.Substring($IndexIntoString, $BreakPoint).Trim() + "`n"
+                            $IndexIntoString += $BreakPoint
+                        }
+                        if ($IndexIntoString -lt $length) {
+                            $folded += $padding + $workingString.Substring($IndexIntoString).Trim() + "`n"
+                        } else {
+                            $folded += "`n"
+                        }
+                    }
+                    $folded
+                } else {
+                        ($string.StartsWith('#'))?"'$($string -replace '''', '''''')'":$string
+                }
+                break
+            }
+
+            'Dictionary`2' {
+                    ("$($InputObject.GetEnumerator() | ForEach-Object {
+                                "`n$padding  $($_.Key): " +
+                                (ConvertTo-PodeYamlInternal -InputObject $_.Value -depth $Depth -NestingLevel ($NestingLevel + 1))
+                            })")
+            }
+            default {
+                "'$InputObject'"
+            }
```

Comment:
Couple points on the above:

* Avoid using `ForEach-Object` here, since we're in the scope of a web request and ForEach-Object is slow you'll want to use a proper `foreach ($item in $items) {}` instead.
* I would highly recommend using a StringBuilder instead of all the string concatenations, it'll be much faster.

This'll be fine as is as a fallback when PSYaml/powershell-yaml aren't installed. If needed, I might port it over to C# just give it some extra welly.

Reply:
done

Reply:
I'm still seeing the `ForEach-Object`s on my side 🤔

Reply:
I forgot one of them